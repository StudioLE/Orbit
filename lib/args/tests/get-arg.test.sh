#!/bin/bash
set -uo pipefail

# ARGS
ACTUAL_FILE=$1

# SETUP

source /srv/lib/args/get-args.sh
source /srv/lib/args/get-arg.sh

get-args \
  --string "Value of --string" \
  --int -1 \
  --flag \
  --optional value \
  -s "Value of -s" \
  -a \
  -bc \
  -- \
  /path/to/file \
  path/to/file

# START

declare OUTPUT

OUTPUT=""

STRING=$(get-arg "--string")
OUTPUT+="--string: ${STRING}"
OUTPUT+=$'\n'

FLAG=$(get-arg "--flag")
OUTPUT+="--flag: ${FLAG}"
OUTPUT+=$'\n'

NOT_SET=$(get-arg "--not-set")
OUTPUT+="--no-set: ${NOT_SET}"
OUTPUT+=$'\n'

ZERO=$(get-arg "0")
OUTPUT+="0: ${ZERO}"
OUTPUT+=$'\n'

A=$(get-arg "-a")
OUTPUT+="-a: ${A}"

echo "${OUTPUT}" > "${ACTUAL_FILE}"
