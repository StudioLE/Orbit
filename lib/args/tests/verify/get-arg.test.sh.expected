--string: Value of --string
--flag: TRUE
--no-set: 
--no-set: OVERRIDE DEFAULT
0: /path/to/file
-a: TRUE
